### PR TITLE
adds fill() functionality in webgl mode; parity with 2d canvas mode

### PR DESCRIPTION
**Pull request by indefinit**
Should clear up some confusion with beginning users.

### CODE REVIEW

**Comment by indefinit** [comment]
whoops looks like this is an immediate mode breaking change.  Standby until I fix it all around.

File: src/3d/immediateMode3D.js [hunk 1]
```diff
@@ -76,19 +77,19 @@ p5.Renderer3D.prototype.quad = function
...
+  this.vertexStack.push(x, y, z);
   return this;
 };
 
 p5.Renderer3D.prototype.endShape = function(){
   var gl = this.GL;
-  this._primitives2D(this.verticeStack);
-  this.verticeStack = [];
+  this._primitives2D(this.vertexStack);
+  this.vertexStack = [];
```
Comment (lmccart) [inline src/3d/immediateMode3D.js]: super minor, I wonder if we should be creating new arrays here, or just reseting the length to 0 when we want to empty?

Comment (indefinit) [inline src/3d/immediateMode3D.js]: Resetting to 0 length could work.  Ill add the change

**Comment by lmccart** [comment]
great. is this one ready to go otherwise or are we still standing by?

**Comment by indefinit** [comment]
Let's standby on this merge because I have a larger change list in the pipe
that will remove a bunch of code from immediateMode3D anyway.

On Sun, Nov 1, 2015 at 9:34 AM, Lauren McCarthy <notifications@github.com>
wrote:

> great. is this one ready to go otherwise or are we still standing by?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/processing/p5.js/pull/1046#issuecomment-152830511>.
>



-- 

Kevin Siwoff
// http://kevinsiwoff.com // Instagram: @kevinsiwoff
<http://instagram.com/kevinsiwoff/> //

**Pull request closed without merging**